internal/kube/client.go

```diff
@@ -40,17 +44,26 @@ func NewInClusterRESTClientGetter(cfg *rest.Config, namespace string) genericcli
 type MemoryRESTClientGetter struct {
 	kubeConfig []byte
 	namespace  string
+	cfg        *rest.Config
 }
 
-func NewMemoryRESTClientGetter(kubeConfig []byte, namespace string) genericclioptions.RESTClientGetter {
+func NewMemoryRESTClientGetter(kubeConfig []byte, namespace string, cfg *rest.Config) genericclioptions.RESTClientGetter {
 	return &MemoryRESTClientGetter{
 		kubeConfig: kubeConfig,
 		namespace:  namespace,
+		cfg:        cfg,
```

Comment:
Instead of embedding the `cfg` here, I think it would be better to inject the (string) values we want to (optionally) overwrite. This would create a better boundary and prevent from e.g. accidentally returning `c.cfg` instead of `cfg` (and thereby leaking a cluster-wide config).

Reply:
I've replaced it with `impersonateAccount`.